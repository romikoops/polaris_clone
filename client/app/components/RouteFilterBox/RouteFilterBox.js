import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import 'react-day-picker/lib/style.css';
import { moment } from '../../constants';
import { RoundButton } from '../RoundButton/RoundButton';
import styles from './RouteFilterBox.scss';
export class RouteFilterBox extends Component {
    constructor(props) {
        super(props);
        this.state = {
            selectedDay: moment().format('DD-MM-YYYY'),
            selectedOption: this.props.moT
        };
        this.editFilterDay = this.editFilterDay.bind(this);
        this.handleOptionChange = this.handleOptionChange.bind(this);
        this.setFilterDuration = this.setFilterDuration.bind(this);
    }
    editFilterDay(event) {
        this.props.setDepartureDate(event.day);
    }
    handleOptionChange(changeEvent) {
        this.setState({
            selectedOption: changeEvent.target.value
        });
        this.props.setMoT(changeEvent.target.value);
    }
    setFilterDuration(event) {
        const dur = event.target.value;
        this.props.setDurationFilter(dur);
    }
    render() {
        const { theme } = this.props;
        return (
            <div
                className={`${
                    styles.filterbox
                } flex-100 layout-row layout-wrap`}
            >
                <div className={styles.pickup_date}>
                    <div className="">
                        <p> Pickup date </p>
                        <DayPickerInput
                            placeholder="DD/MM/YYYY"
                            format="DD/MM/YYYY"
                            value={this.state.selectedDay}
                            onDayChange={this.editFilterDay}
                        />
                    </div>
                    <div className={styles.mode_of_transport}>
                        <p>Mode of transport</p>
                        <div className="radio">
                            <label>
                                <input
                                    type="radio"
                                    value="air"
                                    checked={
                                        this.state.selectedOption === 'air'
                                    }
                                    onChange={this.handleOptionChange}
                                />
                                <i className="fa fa-plane" />
                                Air
                            </label>
                        </div>
                        <div className="radio">
                            <label>
                                <input
                                    type="radio"
                                    value="ocean"
                                    checked={
                                        this.state.selectedOption === 'ocean'
                                    }
                                    onChange={this.handleOptionChange}
                                />
                                <i className="fa fa-ship" />
                                Ocean
                            </label>
                        </div>
                    </div>
                    <div className={styles.transit_time}>
                        <p>Transit time</p>
                        <input
                            type="range"
                            value={this.props.durationFilter}
                            onChange={this.setFilterDuration}
                        />
                        <div className={styles.transit_time_labels}>
                            <p>20 days</p>
                            <p>100 days</p>
                        </div>
                    </div>
                    <RoundButton
                        size="small"
                        text="save filter"
                        theme={theme}
                        active
                    />
                </div>
            </div>
        );
    }
}
RouteFilterBox.PropTypes = {
    theme: PropTypes.object,
    setDurationFilter: PropTypes.func,
    setMoT: PropTypes.func,
    setDepartureDate: PropTypes.func,
    durationFilter: PropTypes.number,
    moT: PropTypes.string
};
