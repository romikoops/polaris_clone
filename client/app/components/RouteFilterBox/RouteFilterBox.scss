.DayPickerInput-OverlayWrapper {
	z-index: 9999;
}
:local {
	.filterbox {
		height: fit-content;
		padding: 25px;
		background-color: #FCFCFC;
		box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.17);
	}
	.pickup_date {		
		p {
			margin: 0 0 15px 0;
		}
		margin-bottom: 30px;
	}
	.dpb{
	  height: 36px;
	  width: 160px;
	}
	.dpb_icon {
	  height: 38px;
	  width: 38px;
	  border: 1px solid #F2F2F2;
	  border-right: none;
	  border-radius: 3px 0 0 3px;
	  background-color: white;
	  i {
	    font-size: 20px;
	  }
	}
	.dpb_picker {
	  height: 36px;
	  width: calc(100% - 20px);
	  border: 1px solid #F2F2F2;
	  border-radius: 0 3px 3px 0;
	  background-color: white;
	  padding: 0 9px;
	}
	.mode_of_transport {
		input {
			margin: 5px 0;
		}
		i {
			margin: 0 8px;
			width: 19px;
			text-align: center;
		}
		margin-bottom: 30px;
	}
	.transit_time_labels {
		display: flex;
		justify-content: space-between;
		margin-top: -10px;
		p {
			font-size: 12px;
		}	
	}
	.transit_time {

		// Hide default styles

		input[type=range] {
		  -webkit-appearance: none; /* Hides the slider so that custom slider can be made */
		  width: 100%; /* Specific width is required for Firefox. */
		  background: transparent; /* Otherwise white in Chrome */
		}

		input[type=range]::-webkit-slider-thumb {
		  -webkit-appearance: none;
		}

		input[type=range]:focus {
		  outline: none; /* Removes the blue border. */
		}

		input[type=range]::-ms-track {
		  width: 100%;
		  cursor: pointer;

		  /* Hides the slider so custom styles can be added */
		  background: transparent; 
		  border-color: transparent;
		  color: transparent;
		}


		// Custom styles

		input {
			overflow: hidden;
		}
		
		// 1. Pointer
		
		/* Special styling for WebKit/Blink */
		input[type=range]::-webkit-slider-thumb {
		  -webkit-appearance: none;
		  border: 1px solid #000000;
		  height: 21px;
		  width: 3px;
			margin-top: -4px;
		  background: black;
		  cursor: pointer;
		  box-shadow: 100px 0 0 100px rgb(210, 210, 210);
		}

		// /* Firefox */
		// input[type=range]::-moz-range-thumb {
		//   box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
		//   border: 1px solid #000000;
		//   height: 21px;
		//   width: 3px;
		//   background: black;
		//   cursor: pointer;
		// }

		// /* IE */
		// input[type=range]::-ms-thumb {
		//   box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
		//   border: 1px solid #000000;
		//   height: 21px;
		//   width: 3px;
		//   background: black;
		//   cursor: pointer;
		// }


		// 2. Track

		input[type=range]::-webkit-slider-runnable-track {
		  width: 100%;
		  height: 12px;
		  cursor: pointer;
      background: -webkit-linear-gradient(left, rgb(14, 175, 80), rgb(0, 138, 203)) !important;
		  border-radius: 1.3px;
		  opacity: 0.9;
		}

		input[type=range]:focus::-webkit-slider-runnable-track {
		  opacity: 1;
		}

		// input[type=range]::-moz-range-track {
		//   width: 100%;
		//   height: 8.4px;
		//   cursor: pointer;
		//   box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
		//   background: #3071a9;
		//   border-radius: 1.3px;
		//   border: 0.2px solid #010101;
		// }

		// input[type=range]::-ms-track {
		//   width: 100%;
		//   height: 8.4px;
		//   cursor: pointer;
		//   background: transparent;
		//   border-color: transparent;
		//   border-width: 16px 0;
		//   color: transparent;
		// }
		// input[type=range]::-ms-fill-lower {
		//   background: #2a6495;
		//   border: 0.2px solid #010101;
		//   border-radius: 2.6px;
		//   box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
		// }
		// input[type=range]:focus::-ms-fill-lower {
		//   background: #3071a9;
		// }
		// input[type=range]::-ms-fill-upper {
		//   background: #3071a9;
		//   border: 0.2px solid #010101;
		//   border-radius: 2.6px;
		//   box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
		// }
		// input[type=range]:focus::-ms-fill-upper {
		//   background: #367ebd;
		// }
	}
}