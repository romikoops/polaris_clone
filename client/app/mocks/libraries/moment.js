import moment from 'moment'

require('moment-timezone')

moment.tz.setDefault('Europe/Amsterdam')
