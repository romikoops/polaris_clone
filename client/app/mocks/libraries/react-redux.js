import * as ReactRedux from 'react-redux'

ReactRedux.connect = jest.fn().mockImplementation((mapStateToProps, mapDispatchToProps) => Component => Component)
