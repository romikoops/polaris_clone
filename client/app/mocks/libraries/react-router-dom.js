import * as ReactRouterDom from 'react-router-dom'

ReactRouterDom.withRouter = jest.fn().mockImplementation(x => x)
