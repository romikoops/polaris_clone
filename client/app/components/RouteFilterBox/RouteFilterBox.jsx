import React, { Component } from 'react'
// import styled from 'styled-components'
import DayPickerInput from 'react-day-picker/DayPickerInput'
import { formatDate, parseDate } from 'react-day-picker/moment'
import PropTypes from '../../prop-types'
import '../../styles/day-picker-custom.css'
import { moment } from '../../constants'
import { switchIcon, capitalize } from '../../helpers'
import styles from './RouteFilterBox.scss'
import { TextHeading } from '../TextHeading/TextHeading'
import { Checkbox } from '../Checkbox/Checkbox'

export class RouteFilterBox extends Component {
  constructor (props) {
    super(props)
    this.state = {
      selectedDay: props.departureDate
        ? moment(props.departureDate).format('DD/MM/YYYY')
        : moment().format('DD/MM/YYYY'),
      selectedOption: {
        air: true,
        ocean: true
      }
    }
    this.editFilterDay = this.editFilterDay.bind(this)
    this.handleOptionChange = this.handleOptionChange.bind(this)
    this.setFilterDuration = this.setFilterDuration.bind(this)
  }
  setFilterDuration (event) {
    const dur = event.target.value
    this.props.setDurationFilter(dur)
  }
  editFilterDay (day) {
    this.props.setDepartureDate(day)
  }
  handleOptionChange (changeEvent, target) {
    this.setState({
      selectedOption: {
        ...this.state.selectedOption,
        [target]: changeEvent
      }
    })
    this.props.setMoT(changeEvent, target)
  }
  render () {
    const {
      theme, pickup, shipment, availableMotKeys
    } = this.props
    // const StyledRange = styled.div`
    //   input[type='range']::-webkit-slider-runnable-track {
    //     width: 100%;
    //     height: 12px;
    //     cursor: pointer;
    //     background: -webkit-linear-gradient(
    //       left,
    //       ${theme.colors.primary},
    //       ${theme.colors.secondary}
    //     ) !important;
    //     border-radius: 1.3px;
    //     opacity: 0.9;
    //   }
    // `
    const dayPickerProps = {
      disabledDays: {
        before: new Date(moment()
          .add(7, 'days')
          .format())
      },
      month: new Date(
        moment()
          .add(7, 'days')
          .format('YYYY'),
        moment()
          .add(7, 'days')
          .format('M') - 1
      ),
      name: 'dayPicker'
    }
    const motCheckBoxes = Object.keys(availableMotKeys).map(mKey => (
      <div className="radio layout-row layout-align-none-center" style={{ margin: '2px 0' }}>
        <Checkbox
          onChange={e => this.handleOptionChange(e, mKey)}
          checked={this.state.selectedOption[mKey]}
          theme={theme}
        />
        <label className="flex-none">
          {switchIcon(mKey)}
          {capitalize(mKey)}
        </label>
      </div>
    ))

    return (
      <div className={styles.filterbox}>
        <div className={styles.pickup_date}>
          <div>
            <TextHeading theme={theme} size={4} text={pickup ? 'Pickup Date' : 'Closing Date'} />
          </div>
          <div className={`flex-none layout-row ${styles.dpb}`}>
            <div className={`flex-none layout-row layout-align-center-center ${styles.dpb_icon}`}>
              <i className="flex-none fa fa-calendar" />
            </div>
            <DayPickerInput
              placeholder="DD/MM/YYYY"
              formatDate={formatDate}
              parseDate={parseDate}
              format="DD/MM/YYYY"
              className={styles.dpb_picker}
              value={this.state.selectedDay}
              onDayChange={this.editFilterDay}
              dayPickerProps={dayPickerProps}
            />
          </div>
        </div>
        <div className={styles.haulage}>
          <div>
            <TextHeading theme={theme} size={4} text="Haulage" />
          </div>
          <div className={`${styles.haulage_option} layout-row layout-wrap flex-none`}>
            <div className="flex-100 layout-row layout-align-space-between-center">
              <p className="flex-none five_m">Pre Carriage</p>
              <p className="flex-none five_m">{shipment.has_pre_carriage ? 'Yes' : 'No'}</p>
            </div>
          </div>
          <div className={`${styles.haulage_option} layout-row layout-wrap flex-none`}>
            <div className="flex-100 layout-row layout-align-space-between-center">
              <p className="flex-none five_m">On Carriage</p>
              <p className="flex-none five_m">{shipment.has_on_carriage ? 'Yes' : 'No'}</p>
            </div>
          </div>
        </div>
        <div className={styles.mode_of_transport}>
          <div>
            <TextHeading theme={theme} size={4} text="Mode of transport" />
          </div>
          {motCheckBoxes}
        </div>
        <div>
          <p style={{ fontSize: '12px', marginTop: '0' }}>* Transit time (T/T) not guaranteed</p>
        </div>
        {/* <StyledRange className={styles.transit_time}>
          <TextHeading theme={theme} size={4} text="Estimated Transit Time" />
          <
          <input type="range" value={this.props.durationFilter} onChange={this.setFilterDuration} />
          <div className={styles.transit_time_labels}>
            <p>20 days</p>
            <p>100 days</p>
          </div>
        </StyledRange> */}
      </div>
    )
  }
}
RouteFilterBox.propTypes = {
  departureDate: PropTypes.number,
  theme: PropTypes.theme,
  setDurationFilter: PropTypes.func.isRequired,
  setMoT: PropTypes.func.isRequired,
  setDepartureDate: PropTypes.func.isRequired,
  // durationFilter: PropTypes.number.isRequired,
  pickup: PropTypes.bool,
  shipment: PropTypes.objectOf(PropTypes.any),
  availableMotKeys: PropTypes.objectOf(PropTypes.bool)
}

RouteFilterBox.defaultProps = {
  departureDate: 0,
  theme: 0,
  pickup: false,
  shipment: {},
  availableMotKeys: {}
}

export default RouteFilterBox
