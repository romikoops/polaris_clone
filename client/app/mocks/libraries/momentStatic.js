import moment from 'moment'

require('moment-timezone')

const constantDate = new Date('June 13, 2017 04:41:20 GMT+1:00')

// eslint-disable-next-line
Date = class extends Date {
  constructor () {
    super()

    return constantDate
  }
}

moment.tz.setDefault('Europe/Amsterdam')
