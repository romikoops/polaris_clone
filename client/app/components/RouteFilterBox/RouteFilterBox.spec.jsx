import '../../mocks/libraries/momentStatic'
import * as React from 'react'
import { shallow } from 'enzyme'
import {
  shipment, identity, theme, cargoItems
} from '../../mocks'

import RouteFilterBox from './RouteFilterBox'

const availableMotKeys = {
  foo: 'FOO',
  bar: 'BAR'
}

const propsBase = {
  availableMotKeys,
  cargos: cargoItems,
  departureDate: 0,
  durationFilter: 1,
  pickup: true,
  setDepartureDate: identity,
  setDurationFilter: identity,
  setMoT: identity,
  shipment,
  theme
}

test('shallow rendering', () => {
  expect(shallow(<RouteFilterBox {...propsBase} />)).toMatchSnapshot()
})

test('pickup is false', () => {
  const props = {
    ...propsBase,
    pickup: false
  }
  expect(shallow(<RouteFilterBox {...props} />)).toMatchSnapshot()
})

test('availableMotKeys is empty object', () => {
  const props = {
    ...propsBase,
    availableMotKeys: {}
  }
  expect(shallow(<RouteFilterBox {...props} />)).toMatchSnapshot()
})
