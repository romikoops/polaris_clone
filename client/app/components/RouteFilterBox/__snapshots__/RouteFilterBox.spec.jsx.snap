// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props.pickup is false 1`] = `
<div
  className="filterbox"
>
  <div
    className="pickup_date"
  >
    <div>
      <TextHeading
        size={4}
        text="Closing Date"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-none layout-row dpb"
    >
      <div
        className="flex-none layout-row layout-align-center-center dpb_icon"
      >
        <i
          className="flex-none fa fa-calendar"
        />
      </div>
      <[object Object]
        className="dpb_picker"
        dayPickerProps={
          Object {
            "disabledDays": Object {
              "before": Date {},
            },
            "month": Date {},
            "name": "dayPicker",
          }
        }
        format="DD/MM/YYYY"
        formatDate={[Function]}
        onDayChange={[Function]}
        parseDate={[Function]}
        placeholder="DD/MM/YYYY"
        value={19}
      />
    </div>
  </div>
  <div
    className="haulage"
  >
    <div>
      <TextHeading
        size={4}
        text="Haulage"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="haulage_option layout-row layout-wrap flex-none"
    >
      <div
        className="flex-100 layout-row layout-align-space-between-center"
      >
        <p
          className="flex-none five_m"
        >
          Pre Carriage
        </p>
        <p
          className="flex-none five_m"
        >
          No
        </p>
      </div>
    </div>
    <div
      className="haulage_option layout-row layout-wrap flex-none"
    >
      <div
        className="flex-100 layout-row layout-align-space-between-center"
      >
        <p
          className="flex-none five_m"
        >
          On Carriage
        </p>
        <p
          className="flex-none five_m"
        >
          No
        </p>
      </div>
    </div>
  </div>
  <div
    className="mode_of_transport"
  >
    <div>
      <TextHeading
        size={4}
        text="Mode of transport"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
  <div>
    <p
      style={
        Object {
          "fontSize": "12px",
          "marginTop": "0",
        }
      }
    >
      * Transit time (T/T) not guaranteed
    </p>
  </div>
</div>
`;

exports[`shallow rendering 1`] = `
<div
  className="filterbox"
>
  <div
    className="pickup_date"
  >
    <div>
      <TextHeading
        size={4}
        text="Pickup Date"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-none layout-row dpb"
    >
      <div
        className="flex-none layout-row layout-align-center-center dpb_icon"
      >
        <i
          className="flex-none fa fa-calendar"
        />
      </div>
      <[object Object]
        className="dpb_picker"
        dayPickerProps={
          Object {
            "disabledDays": Object {
              "before": Date {},
            },
            "month": Date {},
            "name": "dayPicker",
          }
        }
        format="DD/MM/YYYY"
        formatDate={[Function]}
        onDayChange={[Function]}
        parseDate={[Function]}
        placeholder="DD/MM/YYYY"
        value={19}
      />
    </div>
  </div>
  <div
    className="haulage"
  >
    <div>
      <TextHeading
        size={4}
        text="Haulage"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="haulage_option layout-row layout-wrap flex-none"
    >
      <div
        className="flex-100 layout-row layout-align-space-between-center"
      >
        <p
          className="flex-none five_m"
        >
          Pre Carriage
        </p>
        <p
          className="flex-none five_m"
        >
          No
        </p>
      </div>
    </div>
    <div
      className="haulage_option layout-row layout-wrap flex-none"
    >
      <div
        className="flex-100 layout-row layout-align-space-between-center"
      >
        <p
          className="flex-none five_m"
        >
          On Carriage
        </p>
        <p
          className="flex-none five_m"
        >
          No
        </p>
      </div>
    </div>
  </div>
  <div
    className="mode_of_transport"
  >
    <div>
      <TextHeading
        size={4}
        text="Mode of transport"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
  <div>
    <p
      style={
        Object {
          "fontSize": "12px",
          "marginTop": "0",
        }
      }
    >
      * Transit time (T/T) not guaranteed
    </p>
  </div>
</div>
`;
